CloudMusicGear/Proxy.cs

```diff
@@ -211,7 +211,21 @@ private static void OnResponse(Session s)
             }
             else
             {
-                LogEntry($"Error Occured: {url}, StatusCode: {responseStatusCode}");
+                //LogEntry($"Error Occured: {url}, StatusCode: {responseStatusCode}");
+                if (url.EndsWith(".mp3") && Config.ForceIp)
+                {
+                    int? ipIndex = null;
+                    try
+                    {
+                        ipIndex = Config.IpAddressList.IndexOf(Config.IpAddress) + 1;
+                        if (ipIndex == Config.IpAddressList.Count) ipIndex = 0;
+                    }
+                    catch
+                    {
+                        if (Config.IpAddressList.Count > 0) ipIndex = 0;
+                    }
+                    if (ipIndex != null) Config.IpAddress = Config.IpAddressList[ipIndex.Value];
```

Comment:
I found... The piece of code here just redirects to next IP once, and it doesn't do anything to solve current 404 error.

For example, you selected three IPs: A, B and C. Current IP in use is A. The song doesn't exist on server A, so in this response, it still return 404 to the app (it doesn't do anything with the failure on hand), and next time it requests another song, it uses server B. In short words, it cycles to the next server for next song for current error.

Am I right?